### PR TITLE
Add tutorial about running code in the editor

### PR DESCRIPTION
**Pull request by wingedadventurer**
Added this tutorial on running code in the editor, ie. using `tool` keyword in scripts since I haven't found any similar tutorial in the docs.

### CODE REVIEW

File: tutorials/misc/running_code_in_the_editor.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+You can use it for doing many things, but it is mostly useful in level design for visually presenting things that are hard to predict ourselves. Here are some use cases:
+
+- If you have a cannon that shoots cannonballs affected by physics (gravity), you can draw cannonball's trajectory in the editor, making level design a lot easier.
+- If you have jumppads with varying jump heights, you can draw the maximum jump height a player would reach if it jumped on one, also making level design easier.
+- If your player doesn't use a sprite, but draws itself using code, you can make that drawing code execute in the editor to see your player.
+
+How to use it
+-------------
+
+To turn a script into a tool, add ``tool`` keyword at the top of your code, right under the ``extends ...`` line.
```
Comment (mhilbrunner) [inline tutorials/misc/running_code_in_the_editor.rst]: I'd like to change this to *at the top of your code*, full stop, and rather change these new examples to have it at the very top. As far as I can see, all of the (admittedly few) examples thus far place the `tool` keyword in the very first line.

File: tutorials/misc/running_code_in_the_editor.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Try it out
+-----------
+
+Add a ``Sprite`` node to your scene and set the texture to Godot icon. Attach and open a script, and change it to this:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    extends Sprite
+    tool
```
Comment (mhilbrunner) [inline tutorials/misc/running_code_in_the_editor.rst]: Please move the tool keyword to the top of the code

File: tutorials/misc/running_code_in_the_editor.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    func _process(delta):
+        if Engine.editor_hint:
+            rotation_degrees += 180 * delta
+        else:
+            rotation_degrees -= 180 * delta
+
+Save the script. Now the object will spin clockwise in the editor, but if you run the game, it will spin counter-clockwise.
+
+.. note:: Codes from other nodes don't run in the editor. Your access to other nodes is limited. You can access the tree and nodes, and their default properties, but you can't access user variables. If you want to do so, other nodes have to run in the editor too. AutoLoad nodes cannot be accessed in the editor at all.
```
Comment (mhilbrunner) [inline tutorials/misc/running_code_in_the_editor.rst]: *Code from other nodes doesn't run in the editor.* (I'd avoid the plural "codes", sounds weird IMO)

File: tutorials/misc/running_code_in_the_editor.rst (hunk 1 above)
Comment (wingedadventurer) [reply tutorials/misc/running_code_in_the_editor.rst]: That makes more sense, I will make all suggested changes.

**Pull request merged**